src/Execution/Arguments/ArgumentSet.php

```diff
@@ -75,12 +76,47 @@ function (Directive $directive): bool {
         return $argumentSet;
     }
 
+    /**
+     * Check for the @rename directive on any of the fields and rename them accordingly.
+     *
+     * @return $this
+     */
+    public function rename(): self
+    {
+        $argumentSet = new self();
+        $argumentSet->directives = $this->directives;
+
+        foreach ($this->arguments as $name => $argument) {
+            if ($argument->value instanceof self) {
+                // recurse to nested inputs
+                $argument->value = $argument->value->rename();
+            }
+
+            if (empty($argument->directives)) {
```

Comment:
No need for that check, the property is always initialized as an empty collection.